api/models/app.models.js

```diff
@@ -8,31 +8,35 @@ exports.fetchCategories = () => {
 
 exports.fetchReviewByID = (review_id) => {
     const queryValuesArr = [review_id]
-    // const queryStr = `SELECT reviews.*, COUNT(comments.review_id) AS comment_count 
-    // FROM reviews 
-    // JOIN comments
-    // ON reviews.review_id = comments.review_id
-    // WHERE reviews.review_id = $${queryValuesArr.length}
-    // GROUP BY reviews.review_id;`;
-
-    const queryStr = `SELECT reviews.review_id, reviews.title, reviews.review_body, reviews.designer, reviews.review_img_url, reviews.votes, reviews.category, reviews.owner, reviews.created_at, COUNT(comments.review_id) AS comment_count 
-    FROM reviews 
-    JOIN comments
-    ON reviews.review_id = comments.review_id
-    WHERE reviews.review_id = $${queryValuesArr.length}
-    GROUP BY reviews.review_id;`;
-
-
-
-    // const queryStr = `SELECT * FROM reviews WHERE review_id = $${queryValuesArr.length}`;
-
-    return db.query(queryStr, queryValuesArr).then(({rows}) => {
-        if (rows.length === 0){
-            return Promise.reject({status: 404, msg: 'Review not found'})
-        } else {
-            rows[0].comment_count = parseInt(rows[0].comment_count)
-            return rows[0];
+    // let queryStr = ''; 
+    
+    return db.query('SELECT review_id FROM comments').then(({rows}) => {
+        let queryStr = '';
+        // loop over the rows array, break out if object matches
+        for (let i = 0; i < rows.length; i++){
+            if (parseInt(rows[i].review_id) === parseInt(review_id)){
+                return queryStr = `SELECT reviews.*, COUNT(comments.review_id) AS comment_count 
+                FROM reviews 
+                JOIN comments
+                ON reviews.review_id = comments.review_id
+                WHERE reviews.review_id = $${queryValuesArr.length}
+                GROUP BY reviews.review_id;`;
+            }
         }
+        return queryStr = `SELECT * FROM reviews WHERE review_id = $${queryValuesArr.length};`
```

Comment:
If it works, then im hesitant to tell you to refactor it, but this is massively overcomplicated the query. 
You should be able to do the whole thing in the main sql query, getting the review id with the where statement and using a left join. 
Im not sure what the first query and loop is doing.